docs/json.rst

```diff
@@ -0,0 +1,160 @@
+JSON domain
+===========
+
+This theme defines a ``json`` domain that supports documenting `JSON schemas
+<https://json-schema.org/specification.html>`__.
+
+Prerequisite
+------------
+
+This extension requires the `PyYAML <https://pypi.org/project/PyYAML/>`__
+package; you must either add it as a dependency directly, or add ``json`` extra
+feature of this package as a dependency:
+
+.. code-block:: shell
+
+   pip install sphinx-immaterial[json]
+
+
+Configuration
+-------------
+
+To use this domain, add :python:`sphinx_immaterial.json_domain` to the list of
+:python:`extensions` in :file:`conf.py` and set the :confval:`json_schemas`
+configuration option to a list of glob patterns specifying the JSON schema
+definition files.
+
+.. code-block:: python
+
+    extensions = [
+        # other extensions...
+        "sphinx_immaterial.json_domain",
+    ]
+
+    json_schemas = [
+        "**/*_schema.yml",
+    ]
+
+.. confval:: json_schemas
+
+   List of :py:obj:`glob` patterns matching JSON schema definition files
+   relative to the documentation source directory.
+
+   Both JSON and YAML format are supported.
+
+   The top-level schema in each file must specify an ``$id``.
+
+.. confval:: json_schema_validate
+
+   Indicates whether to validate the JSON schemas specified by
+   :confval:`json_schemas`.
+
+   Requires that the `jsonschema <https://pypi.org/project/jsonschema/>`__
+   package is installed; you can either add it as a dependency directly, or
+   depend on the ``jsonschema_validation`` extra feature of this package:
+
+   .. code-block:: shell
+
+      pip install sphinx-immaterial[json,jsonschema_validation]
+
+All of the `object-description-options` also apply to this domain.  Any
+top-level schemas have an object type of ``json:schema`` and any members have an
+object type of ``json:subschema``.
+
+Usage
+-----
+
+.. rst:directive:: json:schema
+
+   Generates documentation for a single JSON schema specified by its ``$id``.
+
+   .. note::
+
+      It is not possible to refer to a schema without an ``$id``.
+
+   The ``title`` property is treated as the synopsis, while the ``description``
+   property is interpreted as reStructedText, with a default role of
+   :rst:role:`json:schema`.
```

Comment:
By default role I meant the Sphinx `default-role`, i.e. the effect of using plain back ticks.  I'll revise the text.

Reply:
oh ok.

Reply:
Since intersphinx is already pulling inventory of sphinx-doc.org, you could ref the [`default_role`](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-default_role) confval.

Reply:
Revised.